app/controllers/api/v1/donor_conditions_controller.rb

```diff
@@ -16,12 +16,7 @@ class DonorConditionsController < Api::V1::ApiController
     api :GET, '/v1/donor_conditions', "List all donor conditions."
     param :ids, Array, of: Integer, desc: "Filter by donor condition ids e.g. ids = [1,2,3,4]"
     def index
```

Comment:
If we abstract away too much code, it makes it harder to introduce slight variations later and the code is less flexible. I would break this down in to a couple of helper methods.

    render_and_return_cached_json if params[:ids].blank?
    @donor_conditions = @donor_conditions.where(id: params[:ids].split(",") ) if params[:ids].present?
    render json: @donor_conditions, each_serializer: serializer

    def render_and_return_cached_json
      controller.model.cached_json and return
    end